<a href="https://colab.research.google.com/github/May-o-nnaise/CPSMA-4313-MS/blob/main/Quiz9.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [3]:
from google.colab import auth
auth.authenticate_user()
print('Authenticated')

Authenticated


#Quiz 9

##Import Data Set

In [8]:

%%bigquery --project dataviz-488505
SELECT *
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
LIMIT 5

Query is running:   0%|          |

Downloading:   0%|          |

,trip_id,subscriber_type,bike_id,bike_type,start_time,start_station_id,start_station_name,end_station_id,end_station_name,duration_minutes
0,27528286,Single Trip (Pay-as-you-ride),21422,electric,2022-08-20 14:03:38+00:00,4051,10th/Red River,4051,10th/Red River,102
1,27497723,Local365,21654,electric,2022-08-17 17:04:01+00:00,4051,10th/Red River,4051,10th/Red River,5
2,27561028,Local31,19221,electric,2022-08-23 11:44:05+00:00,4051,10th/Red River,4051,10th/Red River,26
3,27770713,Local31,18101,electric,2022-09-09 15:47:40+00:00,4051,10th/Red River,4051,10th/Red River,32
4,28005953,Local31,19280,electric,2022-09-28 18:55:34+00:00,4051,10th/Red River,4051,10th/Red River,26


##Compute the average time for a trip based on starting point

In [19]:

%%bigquery --project dataviz-488505
SELECT AVG(duration_minutes) AS avg_time, start_station_name
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
WHERE start_station_name = '10th/Red River'
GROUP BY start_station_name

Query is running:   0%|          |

Downloading:   0%|          |

,avg_time,start_station_name
0,31.151388,10th/Red River


I took the average of the minutes duration and the start station name where the station is the 10th Red River and i grouped it by start station name but I think that last part wasnt necessary but i had fun. This combination told me that the average trip from 10th Red River is 31.151388 minutes long.

##Compute how many trips start at each starting point

In [23]:
%%bigquery --project dataviz-488505
SELECT
  COUNT(start_station_name) AS num, start_station_name
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
GROUP BY start_station_name

Query is running:   0%|          |

Downloading:   0%|          |

,num,start_station_name
0,2629,10th/Red River
1,2169,13th/Trinity @ Waterloo Greenway
2,43716,21st/Guadalupe
3,16080,22.5/Rio Grande
4,27097,2nd/Congress
...,...,...
196,1110,Lakeshore & Pleasant Valley
197,1082,Nash Hernandez @ RBJ South
198,264,Rosewood & Angelina
199,1,Eeyore's 2017


To make this one, I copied over the Count one from the last quiz, removed the * and replaced it with start station name then edited it to be where it just groups by start station name. This is also kinda interesting because I didn't limit the data which means that there is only 201 stations that were recorded.

##Which starting point has the most trips

In [27]:
%%bigquery --project dataviz-488505
SELECT
  COUNT(start_station_name) AS num, start_station_name
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
GROUP BY start_station_name
ORDER BY num DESC
LIMIT 1

Query is running:   0%|          |

Downloading:   0%|          |

,num,start_station_name
0,108559,21st/Speedway @ PCL


To make this one I copied the last question and just added order by descending and limited it to one. Naming the count num helped me becaues now i could just type order by num desc and english wise that makes a lot of sense.

Also it looks like 21st speedway at PCL has the most trips at 108559.